### PR TITLE
Add documentation about repair steps

### PR DESCRIPTION
**Pull request by LEDfan**
I needed this functionality, but it was undocumented :smiley: 

Don't hesitate to give grammar/spelling suggestions!

### CODE REVIEW

File: developer_manual/app/repair.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  	<repair-steps>
+  		<install>
+  			<step>OCA\MyApp\Migration\MyRepairStep</step>
+  		</install>
+  	</repair-steps>
+  </info>
+
+
+Repair-step types
+==========================
```
Comment (MariusBluem) [inline developer_manual/app/repair.rst]: Less ==== 😜

File: developer_manual/app/repair.rst [hunk 2]
```diff
@@ -0,0 +1,111 @@
+=======
+Repair Steps
+=======
```
Comment (MariusBluem) [inline developer_manual/app/repair.rst]: More == 😉

**Comment by MariusBluem** [comment]
Se my two style comments above 😁 ... I understand it, even if I am not a developer. But I don't know if it is content-wise correct ... Maybe @MorrisJobke can review it ;)

File: developer_manual/app/repair.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  		$this->logger->warning("Hello world from MyRepairStep!", ["app" => "MyApp"]);
+  	}
+  }
+
+Outputting information
+~~~~~~~~~~~~~~~~~~~~~
+A repair step can generate information while running, using the
+``OCP\Migration\IOutput`` parameter to the ``run`` method.
+Using the ``info`` and ``warning`` methods a message can be shown in the console.
+In orderto show a progressbar, firstly call the ``startProgress`` method.
```
Comment (MorrisJobke) [inline developer_manual/app/repair.rst]: `In order to ...`

File: developer_manual/app/repair.rst [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+The maximum number of steps can be adjusted by passing it as argument to the
+``startProgress`` method. After every step run the ``advance`` method. When all steps are finished run the ``finishProgress``
```
Comment (MorrisJobke) [inline developer_manual/app/repair.rst]: I would write `Once all steps are ...` instead of `When all steps are ...`

**Comment by MorrisJobke** [comment]
Then this is good.

**Comment by LEDfan** [comment]
@MorrisJobke done :smiley:

**Comment by MorrisJobke** [comment]
stable12 d68e6003 c66f4965 a3b2bb1b

**Pull request merged**